src/xarray_regrid/methods/conservative.py

```diff
@@ -161,16 +159,13 @@ def apply_weights(
     coord: Hashable,
     valid_frac: xr.DataArray,
     skipna: bool,
-    non_grid_dims: list[Hashable],
 ) -> tuple[xr.DataArray, xr.DataArray]:
     """Apply the weights to convert data to the new coordinates."""
     coord_map = {f"target_{coord}": coord}
     weights_norm = weights.copy()
 
     if skipna:
         notnull = da.notnull()
-        if non_grid_dims:
```

Comment:
This was the somewhat arbitrary reduction step I had added to the NaN tracking logic before, to consider grid points where any non-grid sample was null. Taking this out allows us to deal with isolated NaNs in a particular timestamp for example.